src/bin/pg_autoctl/file_utils.c

```diff
@@ -469,26 +470,51 @@ path_in_same_directory(const char *basePath, const char *fileName,
 
 
 /*
- * Searches all the directories in the ':' separated pathlist for
+ * search_path_one copies the first entry found in PATH to result. result
+ * should be a buffer of (at least) MAXPGPATH size.
+ * The function returns false and logs an error when it cannot find the command
+ * in PATH.
+ */
+bool
+search_path_one(const char *filename, char *result)
```

Comment:
I think the name of the function should be `search_path_first` here. I though we would need `search_path_single` too with an error case for more than one item found. We might need both and `search_path_one` is confusing? does it mean `first` or `single`?